packages/inngest/src/express.ts

```diff
@@ -20,10 +22,20 @@ export const serve: ServeHandler = (nameOrInngest, fns, opts) => {
     nameOrInngest,
     fns,
     opts,
-    (req: Request, _res: Response) => {
-      const hostname = req.get("host") || req.headers["host"];
-      const protocol = hostname?.includes("://") ? "" : `${req.protocol}://`;
-      const url = new URL(req.originalUrl, `${protocol}${hostname || ""}`);
+    (
+      req: Either<Request, VercelRequest>,
+      _res: Either<Response, VercelResponse>
+    ) => {
+      const hostname = req.headers["host"];
+
+      const protocol = hostname?.includes("://")
+        ? ""
+        : `${req.protocol || "https"}://`;
+
+      const url = new URL(
+        req.originalUrl || req.url || "",
+        `${protocol}${hostname || ""}`
```

Comment:
I think if the hostname is null this will throw an error. Is there a possibility that `req.headers["host"]` is null or is the `|| ""` just defining against typescript types?

```ts
// this throws an error
new URL("/pathname", "https://") 
```

If it's a TS thing I wonder if it may be nicer to conditionally check for hostname and if it's blank, we look for the `Inngest` client's `serveHost`?

Reply:
Good catch!